lib/devise/redirect_to_login_failure.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class RedirectToLoginFailure < Devise::FailureApp
+  def redirect_url
+    session[:redirect_to] = request.env['ORIGINAL_FULLPATH'] unless session[:redirect_to]
```

Comment:
I think you want `session[:redirect_to] ||= request.env['ORIGINAL_FULLPATH']` Don't set this hash key it's already there?